routes/auth.routes.js

```diff
@@ -0,0 +1,70 @@
+const { Router } = require("express");
+const router = new Router();
+//const bcrypt = require("bcryptjs");
+const mongoose = require("mongoose");
+const User = require("../models/User.model");
+const bcryptjs = require("bcryptjs");
+const saltRounds = 10;
+const { isLoggedIn, isLoggedOut } = require("../middleware/route-guard.js");
+
+router.get("/signup", (req, res) => res.render("auth/signup"));
+
+router.post("/signup", (req, res, next) => {
+  const { username, password } = req.body;
+
+  bcryptjs
+    .genSalt(saltRounds)
+    .then((salt) => bcryptjs.hash(password, salt))
+    .then((hashedPassword) => {
+      return User.create({ username, passwordHash: hashedPassword });
```

Comment:
you can redirect the user to the login page after create a new user.
An alternative is to create a session, like you did in the login route and redirect the user to the profile instead